src/vs/workbench/browser/parts/editor/editorCommands.ts

```diff
@@ -99,6 +99,9 @@ export const NEW_EMPTY_EDITOR_WINDOW_COMMAND_ID = 'workbench.action.newEmptyEdit
 export const API_OPEN_EDITOR_COMMAND_ID = '_workbench.open';
 export const API_OPEN_DIFF_EDITOR_COMMAND_ID = '_workbench.diff';
 export const API_OPEN_WITH_EDITOR_COMMAND_ID = '_workbench.openWith';
+// --- Start Positron ---
+export const API_REOPEN_WITH_EDITOR_COMMAND_ID = '_workbench.reopenWith';
```

Comment:
I added a new command and command ID even though this is very similar to `workbench.action.reopenTextEditor`; the two main differences are:

- this is a command for extensions to use, not a core editor action/contribution 
- the original code only reopens in the default text editor and this new command takes the editor ID as an arg

The other option would be to somehow pipe the API command through to call the editor action/contribution? After looking at how the code is structured, I tend to think this is better. The main downside I see is that if something changed in the implementation of `workbench.action.reopenTextEditor` upstream, we don't have a great way to get prompted in this new command we made.